src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -6576,6 +6580,38 @@ static void Main()
             var verifier = CompileAndVerify(comp, expectedOutput: "3");
         }
 
+        [Fact]
+        public void DeconstructionWarnsForSelfAssignment()
```

Comment:
:memo: The `Swap()` test which already existed in this file tests cases where the same variables are used on the left and right sides but not in the same position, and thus is not a self-assignment.